c/examples/simple.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Copyright The LanceDB Authors
+ */
+
+#include <sys/stat.h>
+#include <iostream>
+#include <memory>
+#include <vector>
+#include <arrow/api.h>
+#include <arrow/c/bridge.h>
```

Comment:
Would be cool to have a simple example with nanoarrow: https://arrow.apache.org/nanoarrow/latest/index.html